lxd/instance/drivers/driver_qemu.go

```diff
@@ -3338,6 +3338,14 @@ func (d *qemu) generateQemuConfigFile(cpuInfo *cpuTopology, mountInfo *storagePo
 	// This is used by the lxd-agent in preference to 9p (due to its improved performance) and in scenarios
 	// where 9p isn't available in the VM guest OS.
 	configSockPath, _ := d.configVirtiofsdPaths()
+
+	socketFile, err := listener.(*net.UnixListener).File()
```

Comment:
This can then be moved outside of `generateQemuConfigFile` by just passing the fd returned from `addFileDescriptor`.